x/staking/keeper/liquid_stake.go

```diff
@@ -441,3 +442,34 @@ func (k Keeper) RefreshTotalLiquidStaked(ctx sdk.Context) error {
 
 	return nil
 }
+
+// CheckVestedDelegationInVestingAccount verifies whether the provided vesting account
+// holds a vested delegation for an equal or greater amount of the specified coin
+// at the given block time.
+//
+// Note that this function facilitates a specific use-case in the LSM module for tokenizing vested delegations.
+// For more details, see https://github.com/cosmos/gaia/issues/2877.
+func CheckVestedDelegationInVestingAccount(
+	account vesting.VestingAccount,
+	blockTime time.Time,
+	coin sdk.Coin,
+) bool {
+	// Get the vesting coins at the current block time
+	vestingAmount := account.GetVestingCoins(blockTime).AmountOf(coin.Denom)
+
+	// Note that the "DelegatedVesting" and "DelegatedFree" values
+	// were computed during the last delegation or undelegation operation
+	delVestingAmount := account.GetDelegatedVesting().AmountOf(coin.Denom)
+	delVested := account.GetDelegatedFree()
+
+	// Calculate the new vested delegated coins
```

Comment:
Why do we need to perform any calc here? Shouldn't the vesting type give us the amount of total vested?

Reply:
The LSM module needs the total delegations vested at the current block time to know how many can be tokenized.
Since the `DelegatedVesting` and `DelegatedFree` are lazily computed and therefore not rebalanced according to the new vested coins over time,  we need these extra calculations.

Reply:
Ahh, right we don't have an API for that. Would be nice to have this on the vesting types, but since it's legacy now maybe not worth it.